src/main/java/be/uantwerpen/scicraft/renderer/BohrBlockEntityRenderer.java

```diff
@@ -0,0 +1,435 @@
+	 */
+	public void makeElectrons(int electronCount, MatrixStack matrices, int lightAbove, VertexConsumerProvider vertexConsumerProvider, T blockEntity, float tickDelta) {
+
+		// for the electron-shell distribution, check the NuclidesTable class.
+
+		int currentShell = 1;
+		int electronCounter = 0;
+		for (int el = 0; el < electronCount; el++) {
+
+			int currentNrOfElectrons = calculateNrOfElectrons(currentShell);
+			if (electronCounter == currentNrOfElectrons) {
+				currentShell++;
+				electronCounter = 0;
+			}
+
+			// evenly distribution of electrons around core
+			int electronsOnCurShell = calcPlaceableElectronsOnShell(electronCount, currentShell, currentNrOfElectrons);
+
+			ArrayList<Float> point = calculatePoint(currentShell, blockEntity, tickDelta, electronsOnCurShell, electronCounter);
+			float x = point.get(0);
+			float y = point.get(1);
+			float z = point.get(2);
+
+			matrices.translate(x, y, z);
+			matrices.scale(0.1f, 0.1f, 0.1f);
+
+			MinecraftClient.getInstance().getItemRenderer().renderItem(electron_stack, ModelTransformation.Mode.GROUND, lightAbove, OverlayTexture.DEFAULT_UV, matrices, vertexConsumerProvider, 0);
+
+			matrices.scale(10, 10, 10);
+			matrices.translate(-x, -y, -z);
+
+			electronCounter++;
+		}
+
+//		currentShell = 1;
+//		electronCounter = 0;
+//		for (int el = 0; el < electronCount; el++) {
+//
+//			int currentNrOfElectrons = calculateNrOfElectrons(currentShell);
+//			if (electronCounter == currentNrOfElectrons) {
+//				currentShell++;
+//				electronCounter = 0;
+//			}
+//
+//			for (int i = 0; i < 120; i++) {
+//				float speedMultiplier = 40+20*(currentShell-1);
+//				float radiusMultiplier = 0.1f*(currentShell-1); // multiplier for how much further each new shell is from the nucleus
+//
+//				float speed = (float)(2*Math.PI)/speedMultiplier; // how fast the electrons rotate
+//				float radius = 0.4f+radiusMultiplier; // distance from core, used to calculate the points
+//				float angle = (float)((2*Math.PI/(120)))*(i);
+//
+//				float x = (float)Math.cos(angle)*radius;
+//				float y = (float)Math.sin(angle)*radius;
+//				float z = (float)Math.sin(angle)*radius*Math.min(currentShell-1, 1); // 0 on first shell, z on every other shell.
+//
+//				if (currentShell != 1) {
+//					float rotateAngle = (float)Math.PI/(2f*(currentShell-1));
+//					if (currentShell > 5) {
+//						rotateAngle = (float)Math.PI/(8f*(currentShell-1));
+//					}
+//					ArrayList<Float> new_y_z = rotateAroundXAxis(y, z, rotateAngle);
+//					y = new_y_z.get(0);
+//					z = new_y_z.get(1);
+//				}
+//				matrices.translate(x, y, z);
+//				matrices.scale(0.02f, 0.02f, 0.02f);
+//				MinecraftClient.getInstance().getItemRenderer().renderItem(electron_stack, ModelTransformation.Mode.GROUND, lightAbove, OverlayTexture.DEFAULT_UV, matrices, vertexConsumerProvider, 0);
+//				matrices.scale(50, 50, 50);
+//				matrices.translate(-x, -y, -z);
+//			}
+//
+//
+//			electronCounter++;
+//		}
```

Comment:
Remove commented out code if it is not needed anymore

Reply:
This is needed, it's the basic way of rendering the electron shells, but it's laggy. Commented this untill (if) I can find a better way.